163 disable search while searching

fredkingham opened the pull request:
While the patient is searching for a name/MRN we disable the search button.

<img width="597" alt="Screenshot 2023-02-27 at 18 10 38" src="https://user-images.githubusercontent.com/2175455/221647866-940087b5-537d-4469-a9d6-dc5a45000a80.png">

Commit:
Removes an unused variable from search.js

Commit:
Changes the search button/form to be disabled while waiting for results

In the simple search page, while we are querying the server for search
results, this disabled the search button and form. The form is reenabled
after results are returned of it the user has changed what they want to
search for.

Commit:
Makes use of a previously unused variable in the search controller tests

This only changes the tests.

fredkingham commented:
The failure is solved by https://github.com/openhealthcare/opal/pull/2024

davidmiller commented:
You sure the imports PR is pointing at a branch that will affect this?

Commit:
Changes the search.searching flag js to not reset

In search.js we set a searching flag after they start searching. Once
the results have come back, if the search parameter has not been changed
then do no reenable the form.

fredkingham committed:
Merge branch 'v0.24.0' into 163-disable-search-while-searching

Commit:
Merge branch 'v0.24.0' into 163-disable-search-while-searching

davidmiller committed:
Fixup: explain the change more accurately

davidmiller commented on opal/core/search/templates/search.html:
```diff
@@ -10,14 +10,15 @@ <h2>
                      slash-key-focus="!state || state==='normal'"
                      ng-blur="enableShortcuts()"
                      ng-focus="disableShortcuts()"
+                     ng-change="searching=false"
```
I think I want a comment here to highlight what this does - e.g. re-enables the search button if you change the search string.
I don't 100% know how to comment syntax inside indented single html element that won't break rendering/parsing

Commit:
Adds a comment to search.html

Adds in a comment to explain what the searching variable does in the
ng-change of the search input.

Commit:
Merge branch '163-disable-search-while-searching' of github.com:openhealthcare/opal into 163-disable-search-while-searching

davidmiller committed:
Merge branch 'v0.24.0' into 163-disable-search-while-searching

davidmiller reviewed (approved):
Observing users, only really seems to matter when a server is slow, but useful then...
